src/hooks/useFocusableInput.tsx

```diff
@@ -0,0 +1,25 @@
+import { useEffect, useRef } from 'react'
+
+export function useFocusableInput(shouldFocus: boolean) {
+  const inputRef = useRef<HTMLInputElement | null>(null)
+  const setInputRef = (instance: HTMLInputElement | null) => {
```

Comment:
@aatbip can you please explain what we are trying to achieve here, doesn't `input` builtin `autofocus` props work?

Reply:
`useFocusableInput` is the hack to autofocus the `TextField` in the `Autocompolete` component because `autofocus` prop doesn't work here since we are removing and attaching the `Autocomplete` element based on the `isOpen` state.